ts/main.ts

```diff
@@ -0,0 +1,99 @@
+import { fetchTeams, Team } from './data';
+
+document.addEventListener('DOMContentLoaded', () => {
+  const teamsTab = document.getElementById('teams-tab') as HTMLButtonElement;
+  const favoritesTab = document.getElementById(
+    'favorites-tab',
+  ) as HTMLButtonElement;
+  const teamsView = document.getElementById('teams-view') as HTMLElement;
+  const detailsView = document.getElementById('details-view') as HTMLElement;
+  const favoritesView = document.getElementById(
+    'favorites-view',
+  ) as HTMLElement;
+  const teamsContainer = document.getElementById(
+    'teams-container',
+  ) as HTMLElement;
+  const clubDetails = document.getElementById('club-details') as HTMLElement;
+  const backButton = document.getElementById(
+    'back-button',
+  ) as HTMLButtonElement;
+
+  const favorites: Team[] = [];
+
+  teamsTab.addEventListener('click', () => {
+    teamsView.classList.add('active');
+    detailsView.classList.remove('active');
+    favoritesView.classList.remove('active');
+  });
+
+  favoritesTab.addEventListener('click', () => {
+    teamsView.classList.remove('active');
+    detailsView.classList.remove('active');
+    favoritesView.classList.add('active');
+    renderFavorites();
+  });
+
+  backButton.addEventListener('click', () => {
+    teamsView.classList.add('active');
+    detailsView.classList.remove('active');
+  });
+
+  function renderTeams(teams: Team[]): void {
+    teamsContainer.innerHTML = '';
+    teams.forEach((team) => {
+      const teamElement = document.createElement('div');
+      teamElement.className = 'team';
+      teamElement.innerHTML = `
+        <div class="team-logo">
+          <img src="${team.WikipediaLogoUrl}" alt="${team.Name}" />
+        </div>
+        <p>${team.Name}</p>
+      `;
```

Comment:
You should not use `innerHTML`. It could cause vulnerability issues if someone were to find a way to modify this. You should be creating these elements with `document.createElemet` and assigning attributes with `setAttribute` and text with `textContent`